source/common/upstream/upstream_impl.cc

```diff
@@ -200,9 +200,10 @@ bool updateHealthFlag(const Host& updated_host, Host& existing_host, Host::Healt
 HostVector filterHosts(const std::unordered_set<HostSharedPtr>& hosts,
                        const std::unordered_set<HostSharedPtr>& excluded_hosts) {
   HostVector net_hosts;
+  net_hosts.reserve(hosts.size());
 
   std::set_difference(hosts.begin(), hosts.end(), excluded_hosts.begin(), excluded_hosts.end(),
-                      net_hosts.begin());
+                      std::inserter(net_hosts, net_hosts.begin()));
```

Comment:
Why was this needed?

Reply:
Without this tests were failing from what seemed to be insertions after the end of the vector. Guessing this was a bug that wasn't caught in the previous PR for some reason, which is strange since this code would have been exercised.